dotnet/src/Connectors/Connectors.OpenAI/AzureSdk/OpenAIChatMessageContent.cs

```diff
@@ -55,7 +56,7 @@ internal OpenAIChatMessageContent(AuthorRole role, string? content, string model
     /// <summary>
     /// A list of the tools called by the model.
     /// </summary>
-    public IReadOnlyList<ChatCompletionsToolCall> ToolCalls { get; }
+    public IReadOnlyList<ChatCompletionsFunctionToolCall> ToolCalls { get; }
```

Comment:
Isn't this going to be a problem when the Azure SDK adds other kinds of tools?

Reply:
Yes, moved back to use the abstraction, but the serialization is lost. I added a specific metadata property for it but I still don't think this is the best approach, probably should just add a JsonElement to the dictionary object.